### ISSUE TITLE
Plugin does not work from inside Eclipse

### ISSUE THREAD

**Issue by javadch** [OpenAPITools/openapi-generator#509]
I have the following plugin spec in my eClipse POM:
```
<plugin>
  <groupId>org.openapitools</groupId>
  <artifactId>openapi-generator-maven-plugin</artifactId>
  <version>3.1.0</version>
  <executions>
    <execution>
    <phase>generate-resources</phase>
      <goals>
        <goal>generate</goal>
      </goals>
    <configuration>
      <inputSpec>${project.basedir}/src/main/resources/api.yaml</inputSpec>
      <generatorName>spring</generatorName>
    <configOptions>
    <dateLibrary>joda</dateLibrary>
    <sourceFolder>src/gen/java/main</sourceFolder>
    <modelPackage>services.nano.timex.models</modelPackage>
    <apiPackage>services.nano.timex.apis</apiPackage>
    <invokerPackage>services.nano.timex.invokers</invokerPackage>
    <delegatePattern>true</delegatePattern>
    <hideGenerationTimestamp>false</hideGenerationTimestamp>
  </configOptions>
  <library>spring-boot</library>
</configuration>
</execution>
</executions>
</plugin>
```
It generates the output artifacts using the `mvn clean package` command, but does not generate the code form inside Eclipse! The IDE complains about the <execution> node saying:

Description Resource Path Location Type Plugin execution not covered by lifecycle configuration: org.openapitools:openapi-generator-maven-plugin:3.1.0:generate (execution: default, phase: generate-resources) pom.xml /timex-backend-api line 34 Maven Project Build Lifecycle Mapping Problem

How can I fix this?

**Comment by wing328** [comment]
@javadch I could be wrong but I don't think the issue is with the OpenAPI Generator plugin.

Would the following article help?

http://www.eclipse.org/m2e/documentation/m2e-execution-not-covered.html

**Comment by jmini** [comment]
@wing328 : yes and no. Our maven plugin could be a little bit more eclipse IDE friendly by integrating some additional metadata. This way no error would be displayed in Eclipse IDE.

**Comment by wing328** [comment]
@jmini 👌 please share more so that the community can help add the metadata.

**Comment by andreas-eternach** [comment]
> @jmini 👌 please share more so that the community can help add the metadata.

I guess it is described here how to make the plugin compatible. I will give it a try http://www.eclipse.org/m2e/documentation/m2e-making-maven-plugins-compat.html

**Comment by wing328** [comment]
@andreas-eternach thanks! Let me know if you need help with the PR.

**Comment by andreas-eternach** [comment]
probably the file 
        openapi-generator/modules/openapi-generator-maven-plugin/src/main/java/org/openapitools/codegen/plugin/CodeGenMojo.java needs to be updated accordingly.

**Comment by wing328** [comment]
@javadch can you please try the fix by https://github.com/OpenAPITools/openapi-generator/pull/1332 to see if it works for you?

**Comment by andreas-eternach** [comment]
> @javadch can you please try the fix by #1332 to see if it works for you?
> 
> Instructions to fetch the fix:
> 
> ```
> git checkout -b andreas-eternach-master master
> git pull https://github.com/andreas-eternach/openapi-generator.git master
> ```

@javadch after pulling the repo as described by @wing328 you need to
1. cd modules/openapi-generator-maven-plugin
2. mvn install
3. in your eclipse-maven-project update the plugin verion to 3.3.2-SNAPSHOT
4. in eclipse update your eclipse-maven-project by pressing Alt-F5

**Comment by wing328** [comment]
@javadch the fix by @andreas-eternach has been merged into master. Please give it a try to see if it works for you.

Thanks @andreas-eternach for contributing the fix.
